src/command.ts

```diff
@@ -131,7 +122,7 @@ export default abstract class Command {
   protected static _globalFlags: Interfaces.FlagInput
 
   static get globalFlags(): Interfaces.FlagInput {
-    return this._globalFlags
+    return Object.assign(this._globalFlags ?? {}, this.enableJsonFlag ? {...jsonFlag} : {})
```

Comment:
If you merge the json flag here, the order in which `enableJsonFlag` and `globalFlags` are set will matter - which we don't want, see #449